src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -39,6 +81,59 @@ public EnumGroupAndName(string group, string name)
         /// <summary>
         /// Gets the name.
         /// </summary>
-        public string Name { get; }
+        public string Name
+        {
+            get
+            {
+                return GetDisplayName();
+            }
+        }
+
+        // Equals and GetHashCode must be overloaded to accommodate the _nameFunc
+        public override bool Equals(object obj)
+        {
+            if (!(obj is EnumGroupAndName))
+            {
+                return false;
+            }
+
+            var second = (EnumGroupAndName)obj;
+
+            return string.Equals(Group, second.Group) && string.Equals(Name, second.Name);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashcode = HashCodeCombiner.Start();
+
+            hashcode.Add(Group);
+            hashcode.Add(Name);
+
+            return hashcode;
+        }
+
+        private string GetDisplayName()
+        {
+            if (_fieldInfo == null)
+            {
+                return _name;
+            }
+            else
+            {
+                if (_displayAttribute != null)
+                {
+                    // Note [Display(Name = "")] is allowed.
```

Comment:
Looks out of place because the returned value isn't passed to the localizer if it's `string.Empty`. Suggest adding something like `... but we will not attempt to localize and empty name`.